lib/modules/manager/sveltos/extract.ts

```diff
@@ -0,0 +1,89 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
+import { trimTrailingSlash } from '../../../util/url';
+import { parseYaml } from '../../../util/yaml';
+import { DockerDatasource } from '../../datasource/docker';
+import { HelmDatasource } from '../../datasource/helm';
+import { isOCIRegistry, removeOCIPrefix } from '../helmv3/oci';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import {
+  ProfileDefinition,
+  type SveltosHelmSource,
+  type SveltosHelmSpec,
+} from './schema';
+import { 
+  fileTestRegex,
+  removeRepositoryName
+} from './util';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  _config?: ExtractConfig,
+): PackageFileContent | null {
+  // check for argo reference. API version for the kind attribute is used
+  if (fileTestRegex.test(content) === false) {
+    logger.debug(
+      `Skip file ${packageFile} as no projectsveltos.io apiVersion could be found in matched file`,
+    );
+    return null;
+  }
+
+  let definitions: ProfileDefinition[];
+  try {
+    definitions = parseYaml(content, null, {
+      customSchema: ProfileDefinition,
+      failureBehaviour: 'filter',
+      removeTemplates: true,
+    });
+  } catch (err) {
+    logger.debug({ err, packageFile }, 'Failed to parse Sveltos definition.');
+    return null;
+  }
+
+  const deps = definitions.flatMap(processAppSpec);
+
+  return deps.length ? { deps } : null;
+}
+
+function processSource(source: SveltosHelmSource): PackageDependency | null {
+    if (isOCIRegistry(source.repositoryURL) || !source.repositoryURL.includes('://')) {
+      const registryURL = trimTrailingSlash(removeOCIPrefix(source.repositoryURL));
+
+      return {
+        depName: `${registryURL}`,
+        currentValue: source.chartVersion,
+        datasource: DockerDatasource.id,
+      };
+    }
+
+    return {
+      depName: removeRepositoryName(source.repositoryName, source.chartName),
+      registryUrls: [source.repositoryURL],
+      currentValue: source.chartVersion,
+      datasource: HelmDatasource.id,
+    };
+}
+
+function processAppSpec(
+  definition: SveltosHelmSource,
+): PackageDependency[] {
+  const spec: SveltosHelmSpec = definition.spec
+
+  const deps: (PackageDependency | null)[] = [];
+
+  if (is.nonEmptyObject(spec.source)) {
+    deps.push(processSource(spec.source));
+  }
+
+  for (const source of coerceArray(spec.sources)) {
```

Comment:
Where is this sources coming from?
I think this whole function needs a rework.

Reply:
Copied from ArgoCD https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/manager/argocd/extract.ts#L81

Do you have any hints what needs to be changed (I am a ts noob) :D

Reply:
You are trying to access a field which does not exist on the object. 
`spec.source` and `spec.sources` references these fields: 
- https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/manager/argocd/__fixtures__/validApplication.yml#L66
- https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/manager/argocd/__fixtures__/validApplication.yml#L82